byron/ledger/executable-spec/src/Ledger/UTxO/Generators.hs

```diff
@@ -191,20 +200,22 @@ shrinkLeftPreserving inValue outValue modifyOutValue (xs1, ys1) = do
       lost = sum (inValue <$> nodeValue xs1) - sum (inValue <$> nodeValue xs3)
       ys2  = subFromList lost outValue modifyOutValue <$> ys1
     pure $ interleaveInputOutput inValue outValue modifyOutValue (xs3, ys2)
- where
-  subFromList
-    :: Integer
-    -> (a -> Integer)
-    -> ((Integer -> Integer) -> a -> a)
-    -> [a]
-    -> [a]
-  subFromList n getVal modifyVal = go n
-   where
-    go 0  x        = x
-    go _  []       = []
-    go n' (x : xs) = if getVal x > n'
-      then modifyVal (subtract n') x : xs
-      else go (n' - getVal x) xs
+
+-- | Remove total value from a list, removing from the front
+subFromList
+  :: (Num n, Ord n)
+  => n
```

Comment:
You know what I'm going to say: can you add some comments on the function parameters? :pray: